### ISSUE TITLE
Handling avahi-daemon restarts for announced services

### ISSUE THREAD

**Issue by alecpetrosky** [holoplot/go-avahi#9]
It seems that if avahi-daemon has been restarted and therefore service registration is lost, go-avahi does not currently handle such events. Am I right? And if yes, what is the most straightforward way to handle such event in order to update/re-add the service on avahi-daemon?

**Comment by zonque** [comment]
Yes, you're right. The code in the package currently fully relies on the Avahi daemon to persist the client's state, and does not do any extra caching.

A cheap way out would be to inform the user of the package about the connection loss and off-load the work its way. But we could as well cache in the library. For browser functionality, the best thing we can do is to start from the beginning anyway. The only thing that needs attention is the entry group for publishing, right?
